File: 0684/sesión14 - consulta/SoftVen/SoftVenBusiness/ProductoBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftVen.DAO;
using SoftVen.Model;
using SoftVenPersistance.DAOImpl;

namespace SoftVen.BO
{
    public class ProductoBO
    {
        private ProductoDAO productoDAO;

        public ProductoBO()
        {
            productoDAO = new ProductoDAOImpl();
        }

        public int Insertar(string nombre, string descripcion, double precio)
        {
            ProductosDTO productosDTO = new ProductosDTO();
            productosDTO.Nombre = nombre;
            productosDTO.Descripcion = descripcion;
            productosDTO.Precio = precio;

            return this.productoDAO.Insertar(productosDTO);
        }

        public int Modificar(int productoId, string nombre, string descripcion, double precio)
        {
            ProductosDTO productosDTO = new ProductosDTO();
            productosDTO.ProductoId = productoId;
            productosDTO.Nombre = nombre;
            productosDTO.Descripcion = descripcion;
            productosDTO.Precio = precio;
            return this.productoDAO.Modificar(productosDTO);
        }

        public int Eliminar(int productoId)
        {
            ProductosDTO productosDTO = new ProductosDTO();
            productosDTO.ProductoId = productoId;
            return this.productoDAO.Eliminar(productosDTO);
        }

        public ProductosDTO ObtenerPorId(int productoId)
        {
            return this.productoDAO.ObtenerPorId(productoId);
        }

        public BindingList<ProductosDTO> ListarTodos()
        {
            return this.productoDAO.ListarTodos();
        }
    }
}

File: 0684/sesión18 - WS - SOAP/Back-End/SoftVenBusiness/src/main/java/pe/edu/pucp/softven/bo/ProductoBO.java
package pe.edu.pucp.softven.bo;

import java.util.ArrayList;
import pe.edu.pucp.softven.dao.ProductoDAO;
import pe.edu.pucp.softven.daoImp.ProductoDAOImpl;
import pe.edu.pucp.softven.model.ProductosDTO;

public class ProductoBO {

    private ProductoDAO productoDAO;
    
    public ProductoBO(){
        this.productoDAO = new ProductoDAOImpl();
    }
    
    public Integer insertar(String nombre, String descripcion, Double precio){
        ProductosDTO productosDTO = new ProductosDTO();
        productosDTO.setNombre(nombre);
        productosDTO.setDescripcion(descripcion);
        productosDTO.setPrecio(precio);
        
        return this.productoDAO.insertar(productosDTO);
    }
    
    public ProductosDTO obtenerPorId(Integer productoId){
        return this.productoDAO.obtenerPorId(productoId);
    }
    
    public ArrayList<ProductosDTO> listarTodos(){
        return this.productoDAO.listarTodos();
    }
    
    public Integer modificar(Integer productoId, String nombre, String descripcion, Double precio){
        ProductosDTO productosDTO = new ProductosDTO();
        productosDTO.setProductoId(productoId);
        productosDTO.setNombre(nombre);
        productosDTO.setDescripcion(descripcion);
        productosDTO.setPrecio(precio);
        return this.productoDAO.modificar(productosDTO);
    }
    
    public Integer eliminar(Integer productoId){
        ProductosDTO productosDTO = new ProductosDTO();
        productosDTO.setProductoId(productoId);
        return this.productoDAO.eliminar(productosDTO);
    }
    
}
